client/my-sites/earn/home.tsx

```diff
@@ -262,6 +263,51 @@ const Home: FunctionComponent< ConnectedProps > = ( {
 		};
 	};
 
+	/**
+	 * Return the content to display in the Paid Newsletter card based on the current plan.
+	 *
+	 * @returns {object} Object with props to render a PromoCard.
+	 */
+	const getPaidNewsletterCard = () => {
+		const cta = isFreePlan
+			? {
+					text: translate( 'Unlock this feature' ),
+					action: () => {
+						trackUpgrade( 'any-paid-plan', 'premium-content' );
+						page( `/plans/${ selectedSiteSlug }` );
```

Comment:
Hmm it looks like we have a kind of funny analytics call grouping here, but note that it's possible to actually send out one redux event using the navigation redux event, navigation middleware, and decorating the navigation event with analytics meta.

It looks a bit like:

```jsx
const navigateToPlans = ( selectedSiteSlug ) =>
    withAnalytics(
		composeAnalytics(
			recordGoogleEvent( 'My example event', 'Clicked X button on foo' ),
			recordTracksEvent( 'calypso_example_tracks_event ),
			bumpStat( 'example-group', 'example-stat' )
		),
		navigate( `/plans/${ selectedSiteSlug }/` )
	);

const mapDispatchToProps = { navigateToPlans };
const mergeProps = ( stateProps, dispatchProps, ownProps ) => {
    return {
        ...stateProps,
        ...dispatchProps,
        navigateToPlans: () => dispatchProps.navigateToPlans( stateProps.selectedSiteSlug ),
        ...ownProps,
    };
};
```

This one is optional since it looks like this entire file would need the refactor in this case to reduce confustion.